src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+            const parts = name.split(new RegExp(`(${input})`, "gi"));
+
+            return parts
+                .map((part: any, index: number) => {
+                    if (index !== 0) {
+                        return `<span class="sgn-highlight">${part}</span>`;
+                    } else {
+                        return part;
+                    }
+                })
+                .join("");
+        }
+        return name;
+    }
+    _startsWithHighlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (input && name.toLowerCase().startsWith(input.toLowerCase())) {
+            const start = name.substring(0, input.length);
+            const rest = name.substring(input.length);
+            return `<span class="sgn-highlight" >${start}</span>${rest}`;
+        }
+        return name;
+    }
+    setSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more?: boolean
+    ) {
+        this.result = [...suggestions];
+        this.index = this.index;
+        this.input_temp = (this.input_el as HTMLInputElement)?.value;
+
+        this.input_temp ? (this.show_empty = false) : (this.show_empty = true);
+        this.has_more = has_more || false;
+        if (this._private_slot_el) {
+            for (let el of this.private_items) el.remove();
+        }
+        this.size = suggestions.length;
+        for (let i of suggestions) {
+            let tmp = this.template_slots[0].innerHTML;
+            let temp = this._format(tmp, i);
+            let el = document.createElement("div");
+            el.className = "sgn-item";
+            el.slot = "__private-item";
+            el.innerHTML = temp;
+            this.appendChild(el);
+            this._private_slot_el = true;
+        }
+        if (this._private_slot_el && has_more) {
+            let el = document.createElement("div");
+            el.className = "sgn-item";
+            el.slot = "__private-item";
+            el.innerHTML = `<div class="container"><div>Loadmore</div></div>`;
+            this.appendChild(el);
+        }
+    }
+    loadmoreentry(event:Event) {
+        const loadmore = new CustomEvent("loadmore", {
+            bubbles: true,
+        });
+        if (this._private_slot_el) {
+            this.private_items[this.private_items.length - 1].remove();
+        }
+
+        this.dispatchEvent(loadmore);
+        event.stopPropagation();
+    }
+
+    appendSuggestion(
+        suggestions: Array<{ [key: string]: string }>,
+        has_more: boolean
+    ) {
+        this.result = [...this.result];
+        this.result.push(...suggestions);
+        this.result = this.result.flat(Infinity);
+        this.has_more = has_more;
+        this.size = this.size + suggestions.length;
+        for (let i of suggestions) {
+            let tmp = this.template_slots[0].innerHTML;
+            let temp = this._format(tmp, i);
```

Comment:
Same as in setSuggestions. I think you can put together the same part of the code in a common function and call that function from setSuggestions as well as appendSuggestions so that we can avoid duplicate redundant portions of code.